lib/pbench/agent/meister/start.py

```diff
@@ -0,0 +1,438 @@
+            logger.exception("failed to create redis server, daemonized")
+            return 1
+        else:
+            if retcode != 0:
+                logger.error(
+                    "failed to create redis server, daemonized; return code: %d",
+                    retcode,
+                )
+                return 1
+
+        try:
+            timeout = time.time() + REDIS_MAX_WAIT
+            started_channel = "{}-start".format(self.channel)
+            redis_connection_state = "connecting"
+            redis_server = redis.Redis(host="localhost", port=self.redis_port, db=0)
+            pubsub = redis_server.pubsub()
+            while redis_connection_state == "connecting":
+                try:
+                    pubsub.subscribe(started_channel)
+                    chan = pubsub.listen()
+                    # Pull off first message which is an acknowledgement we have
+                    # successfully subscribed.
+                    resp = next(chan)
+                except redis.exceptions.ConnectionError:
+                    if time.time() > timeout:
+                        raise
+                    time.sleep(0.1)
+                else:
+                    redis_connection_state = "connected"
+        except Exception as exc:
+            logger.error(
+                "Unable to connect to redis server, %s:%d: %r",
+                "localhost",
+                self.redis_port,
+                exc,
+            )
+            return self.kill_redis_server(redis_pid)
+        else:
+            assert resp["type"] == "subscribe", f"bad type: f{resp!r}"
+            assert resp["pattern"] is None, f"bad pattern: {resp!r}"
+            assert (
+                resp["channel"].decode("utf-8") == started_channel
+            ), f"bad channel: {resp!r}"
+            assert resp["data"] == 1, f"bad data: {resp!r}"
+
+        # 3. Start the tool-data-sink process
+        #   - leave a PID file for the tool data sink process
+        #   - FIXME: use podman to start a tool-data-sink container
+        tds_param_key = "tds-{}".format(self.context.group)
+        tds = dict(channel=self.channel, benchmark_run_dir=benchmark_run_dir)
+        try:
+            redis_server.set(tds_param_key, json.dumps(tds, sort_keys=True))
+        except Exception:
+            logger.exception(
+                "failed to create tool data sink parameter key in redis server"
+            )
+            return self.kill_redis_server(redis_pid)
+        data_sink = "pbench-tool-data-sink"
+        data_sink_path = _prog_dir / data_sink
+        logger.debug("starting tool data sink")
+        try:
+            retcode = os.spawnl(
+                os.P_WAIT,
+                data_sink_path,
+                data_sink,
+                "localhost",
+                str(self.redis_port),
+                tds_param_key,
+            )
+        except Exception:
+            logger.exception("failed to create pbench data sink, daemonized")
```

Comment:
What is _daemonized_ supposed to indicate here?

Reply:
Is this message a hold over from the existing code?  See https://github.com/distributed-system-analysis/pbench/blob/master/agent/util-scripts/pbench-tool-meister-start#L368

Reply:
straight copy with toolmeister